### ISSUE TITLE
Capture Dyanamic Dependencies

### ISSUE THREAD

**Issue by KGrewal1** [facebook/buck2#432]
I am trying to create rules for Fortran, but I can't work out how to capture the .mod files produced by compilation, as they are unknown until that point. This seems possible with dynamic_output, but I can't seem to work out how.

**Comment by ndmitchell** [comment]
Can you write down (or link to) the pseudo-code for what the ideal compilation flow would be? I'm aware Fortran is very hard, but not that aware of the details. Do you know how Bazel does it? In general Buck2 has more features than Bazel, so if its possible with Bazel, that approach should also work with Buck2.

**Comment by KGrewal1** [comment]
https://github.com/KGrewal1/disconnect can see my vague attempts here: fortran produces .mod files along with the actual output of the compilation and I want to try to write a rule that can capture all of these that are output, without knowing necessarily what they are going to be before compilation.
Unfortunately I cannot find bazel rules for fortran either

**Comment by ndmitchell** [comment]
I suggest we try and figure out how to do this in the most dumb way possible, then iterate to make it better. Starting with a binary and then iterating to make a library second also seems smart.

What would be the correct sequence of commands to compile your two examples, if you were doing it by hand? What files would each produce? How do you figure out the sequence?

**Comment by KGrewal1** [comment]
I agree, that's my current intent.  'gfortran -O3 -o disconnection disconnectionDPS.f90' works by hand but the input file has modules in its structure which out then output to the root of the directory as opposed to with the target, using buck2. For the other file in the repo, a similar instruction works as there aren't submodules so that seems like some sort of start

**Comment by ndmitchell** [comment]
So I guess you need to find the inputs (which shouldn't be hard, assuming they are given explicitly in the BUCK file), but also make sure the .mod outputs don't go anywhere untoward. Should you be copying all the inputs to a temporary directory, and running a change directory, so all the modules get put in a sensible place?
